keanu-project/src/main/java/io/improbable/keanu/network/NetworkSnapshot.java

```diff
@@ -1,44 +1,36 @@
 package io.improbable.keanu.network;
 
-import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.algorithms.variational.optimizer.Variable;
 
-import java.util.*;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
 
 /**
  * Saves the state (value and observed) of a specified collection of vertices.
  */
 public class NetworkSnapshot {
 
-    public static NetworkSnapshot create(Set<Vertex> vertices) {
+    public static NetworkSnapshot create(Set<? extends Variable> vertices) {
```

Comment:
Can we chat about this offline? This makes sense but we might be able to avoid needing to go this route.